api/circulation.py

```diff
@@ -1634,6 +1634,14 @@ def run(self):
                     "Synced %s in %.2f sec", self.api.__class__.__name__, after - before
                 )
 
+                # While testing we are in a Session scope
+                # we need to only close this if api._db is a flask_scoped_session.
+                if getattr(self.api, "_db", None) and type(self.api._db) != Session:
```

Comment:
I wonder if it would be better to not do this check here, but create a `db_close()` function that we can mock out when we are calling tests, so we're sure that our sessions all get closed?

Reply:
I can mock the scoped session close() method itself, I think that would work.

Reply:
While doing this I can see dozens of places this will need to be mocked, making this a little messy.
Additionally, it would be confusing while writing new tests and seeing them fail for no apparent reason.